ONSdigital/eq-author
# 694 validation metadata earliest latest date

### What is the context of this PR?

Implements functionality which allows users to validate the earliest and latest date against metadata.

### How to review 
- Run tests
- Ensure feature works as per https://trello.com/c/rdPAy7Bm/477-validation-against-metadata-for-earliest-date-4 and https://trello.com/c/FAmXFiYB/676-validation-against-metadata-for-latest-date-3

### Dependencies

- [x] https://github.com/ONSdigital/eq-author/pull/475
- [x] https://github.com/ONSdigital/eq-author-api/pull/166
